File: Behavorial/Strategy/strategy.java
class Context{
    int value;
    Strategy s;
    
    Context(int value,Strategy s){
        this.value=value;
        this.s=s;
    }
    
    void calculate(){
        s.compute(value);
    }
    
}

interface Strategy{
    public void compute(int value);
}

class Strategy1 implements Strategy{
    public void compute(int value){
        if(value<40)
            System.out.println("Slim");
        else
            System.out.println("Fat");
            
    }
}

class Strategy2 implements Strategy{
    public void compute(int value){
        if(value<60)
            System.out.println("Slim");
        else
            System.out.println("Fat");
            
    }
}

class Strategy3 implements Strategy{
    public void compute(int value){
        if(value<90)
            System.out.println("Slim");
        else
            System.out.println("Fat");
            
    }
}

public class ComputePhysique{

     public static void main(String []args){
     
     Strategy s1=new Strategy1();
     Strategy s2=new Strategy2();
     Strategy s3=new Strategy3();
     
     Context c1=new Context(60,s1);
     c1.calculate();
     
     Context c2=new Context(60,s2);
     c2.calculate();
     
     Context c3=new Context(60,s3);
     c3.calculate();
     
     }
     
}

File: Behavorial/Strategy/strategy.py
import abc

class Context(object):
    def __init__(self,value,strategy):
        self._value=value
        self._strategy=strategy

    def calculate(self):
        self._strategy.compute(self._value)

class Strategy(metaclass=abc.ABCMeta):
    @abc.abstractmethod
    def compute(self,val):
        pass

class Strategy1(Strategy):
    def compute(self,value):
        if value<40:
            print("Slim")
        else:
            print("Fat")

class Strategy2(Strategy):
    def compute(self,value):
        if value<60:
            print("Slim")
        else:
            print("Fat")

class Strategy3(Strategy):
    def compute(self,value):
        if value<90:
            print("Slim")
        else:
            print("Fat")            


if __name__ == '__main__':
    s1=Strategy1()
    s2=Strategy2()
    s3=Strategy3()

    c1=Context(60,s1)
    c1.calculate()

    c2=Context(60,s2)
    c2.calculate()

    c3=Context(60,s3)
    c3.calculate()
